salt/modules/zypperpkg.py

```diff
@@ -1700,28 +1718,41 @@ def upgrade(refresh=True,
             cmd_update.extend(['--from' if dist_upgrade else '--repo', repo])
         log.info('Targeting repos: %s', fromrepo)
 
-    if dist_upgrade:
-        # TODO: Grains validation should be moved to Zypper class
-        if __grains__["osrelease_info"][0] > 11:
-            if novendorchange:
-                cmd_update.append("--no-allow-vendor-change")
-                log.info("Disabling vendor changes")
+    # TODO: Grains validation should be moved to Zypper class
+    if novendorchange:
+        if dist_upgrade:
+            if __grains__["osrelease_info"][0] > 11:
+                if novendorchange:
```

Comment:
I'm not sure if this makes sense. `novendorchange' is the default for SLE if I'm not mistaken. So if you set this to `False` we don't check if this is a dist-upgrade and also won't check if `novendorchange` is `False` to allow the vendor change then. Did I miss something?